src/setup-beam.js

```diff
@@ -602,11 +602,11 @@ function parseVersionFile(versionFilePath0) {
   // If we ever start parsing something else, this should
   // become default in a new option named e.g. version-file-type
   versions.split('\n').forEach((line) => {
-    const appVersion = line.match(/^([^ ]+)[ ]+([^ #]+)/)
+    const appVersion = line.match(/^([^ ]+)[ ]+(ref:v)?([^ #]+)/)
```

Comment:
The `v` should probably be optional, right?

Other `ref:` might be `ref:39cb398vb39` as explained in [Configuration | asdf](https://asdf-vm.com/manage/configuration.html#tool-versions).